worlds/overcooked2/__init__.py

```diff
@@ -0,0 +1,509 @@
+
+        # Make Items
+        # useful = list()
+        # filler = list()
+        # progression = list()
+        for item_name in item_table:
+            if not self.options["IncludeHordeLevels"] and item_name in ["Calmer Unbread", "Coin Purse"]:
+                # skip items which are irrelevant to the seed
+                continue
+            
+            if not self.options["KevinLevels"] and item_name.startswith("Kevin"):
+                continue
+
+            if is_item_progression(item_name, self.level_mapping, self.options["KevinLevels"]):
+                # print(f"{item_name} is progression")
+                # progression.append(item_name)
+                classification = ItemClassification.progression
+            else:
+                # print(f"{item_name} is filler")
+                if (is_useful(item_name)):
+                    # useful.append(item_name)
+                    classification = ItemClassification.useful
+                else:
+                    # filler.append(item_name)
+                    classification = ItemClassification.filler
+
+            if item_name in item_frequencies:
+                freq = item_frequencies[item_name]
+
+                while freq > 0:
+                    self.itempool.append(self.create_item(item_name, classification))
+                    classification = ItemClassification.useful  # only the first progressive item can be progression
+                    freq -= 1
+            else:
+                self.itempool.append(self.create_item(item_name, classification))
+
+        # print(f"progression: {progression}")
+        # print(f"useful: {useful}")
+        # print(f"filler: {filler}")
+
+        # Fill any free space with filler
+        pool_count = len(oc2_location_name_to_id)
+        if not self.options["KevinLevels"]:
+            pool_count -= 8
+
+        while len(self.itempool) < pool_count:
+            self.itempool.append(self.create_item("Bonus Star", ItemClassification.useful))
+
+        self.world.itempool += self.itempool
+
+    def set_rules(self):
+        pass
+
+    def generate_basic(self) -> None:
+        # Add Events (Star Acquisition)
+        for level in Overcooked2Level():
+            if not self.options["KevinLevels"] and level.level_id > 36:
+                break
+
+            if level.level_id != 36:
+                self.place_event(level.location_name_level_complete, level.event_name_level_complete)
+
+            if self.is_level_horde(level.level_id):
+                continue  # horde levels don't have star rewards
+
+            for n in [1, 2, 3]:
+                self.place_event(level.location_name_star_event(n), "Star")
+
+        # Add Victory Condition
+        self.place_event("6-6 Completed", "Victory")
+
+    # Items get distributed to locations
+
+    def fill_json_data(self) -> Dict[str, Any]:
+        mod_name = f"AP-{self.world.seed_name}-P{self.player}-{self.world.player_name[self.player]}"
```

Comment:
There's a new `self.world.get_out_file_name_base(self.player)` that puts all this info together in a file name.
I don't know if that would work here.